src/jit/hwintrinsicxarch.cpp

```diff
@@ -218,6 +228,43 @@ int Compiler::numArgsOfHWIntrinsic(NamedIntrinsic intrinsic)
     return hwIntrinsicInfoArray[intrinsic - NI_HW_INTRINSIC_START - 1].numArgs;
 }
 
+//------------------------------------------------------------------------
+// numArgsOfHWIntrinsic: get the number of arguments based on baseType
+//
+// Arguments:
+//    intrinsic -- id of the intrinsic function.
+//    baseType  -- var_types type of intrinsic
+//
+// Return Value:
+//     number of arguments
+//
+int Compiler::numArgsOfHWIntrinsic(NamedIntrinsic intrinsic, var_types baseType)
+{
+    assert(intrinsic != NI_Illegal);
+    assert(intrinsic > NI_HW_INTRINSIC_START && intrinsic < NI_HW_INTRINSIC_END);
+
+    switch (intrinsic)
+    {
+        case NI_SSE2_Shuffle:
+            assert(hwIntrinsicArgsInfoArray[0].intrinsicID == intrinsic);
+            assert((baseType >= TYP_BYTE) && (baseType <= TYP_DOUBLE));
+            return hwIntrinsicArgsInfoArray[0].numArgs[baseType - TYP_BYTE];
+        // case NI_SSE_ReciprocalScalar:
+        // case NI_SSE_ReciprocalSqrtScalar:
+        // case NI_SSE_SqrtScalar:
+        // case NI_SSE41_CeilingScalar:
+        // case NI_SSE41_FloorScalar:
+        // case NI_SSE41_RoundCurrentDirectionScalar:
+        // case NI_SSE41_RoundToNearestIntegerScalar:
+        // case NI_SSE41_RoundToNegativeInfinityScalar:
+        // case NI_SSE41_RoundToPositiveInfinityScalar:
+        // case NI_SSE41_RoundToZeroScalar:
```

Comment:
Why are these here in comments? Either add an explanation or remove these comments, please.